### DISCUSSION TITLE
A developer can add 'Tech Stack' in profile

### DISCUSSION THREAD

**Discussion by ghost** [joemasilotti/railsdevs.com#702] [General]
**Why is this valuable?** - Rails developers use various technologies for frontend, data storage, automation, devops, etc. and it might be a value addition to provide them something where they can add these technologies so that hiring managers can see which technologies is this developer well-versed or comfortable with. I found a text field as a simple and an appropriate solution to start with and later on this feature can be further developed where the technologies are saved in the DB and kinda work like tags.

**How it helps developers get hired?** - Hiring managers can see which technologies a developer is well-versed or comfortable with and this might be beneficial for them to select a more matching candidate.

**What workarounds or alternatives exist?** - A workaround is to enter the tech-stack in 'Bio' section which currently I am using for my profile ([https://railsdevs.com/developers/849](https://railsdevs.com/developers/849)).

I have submitted a pull request #698 for your kind perusal.

**Comment by joemasilotti** [comment]
I see businesses searching for developers with experience in different technologies. So I can definitely see the value in this. And your workaround, putting it in the bio, works but isn't great. A more structured way to add and filter could help in finding the right person (and getting hired!).

One concern is that this might quickly become a "keyword stuffing" situation. What stops someone from adding 50 different technologies so they show up in more search results? How do we ensure developers only add a few? And where do we cut them off?

Second, I feel like we can get more specific than "technologies" or "tech stack". I work every day on a Unix system, but I am no expert in it. I also use Postgres for every Rails app but I barely know the difference between that and MySql. I'd rather see this branded as "expertise" or "interests". That might better show intent for someone who knows that tech inside-and-out or is excited to work with (and/or learn) it on the job.

What do you think?

**Comment by ghost** [comment]
Regarding concerns,

First - textfield can be validated with max chars with placeholder saying
`what can be entered` - let's say 150
Second - "Expertise" sounds better as it would help solving the first
concern too to some extent

What do you think?

**Reply by joemasilotti** [reply]
I'm liking this idea more the more I think about it. But I still worry that having a freeform field will end up with results like "Turbo" and "Hotwire" forcing businesses to search for both.

I'm leaning more towards a set list of tags and such that folks could select up to 5 of. But I still want to get feedback from other developers.

Have you heard from anyone in your network wanting something like this?
